lib/ExecutionContext.js

```diff
@@ -94,8 +94,23 @@ class ExecutionContext {
     if (typeof pageFunction !== 'function')
       throw new Error('The following is not a function: ' + pageFunction);
 
+    let functionText = pageFunction.toString();
+    try {
+      eval(functionText);
+    } catch (e1) {
+      // This means we have a function shorthand. Try another
+      // time prefixing 'function '.
+      functionText = 'function ' + functionText;
```

Comment:
What about async static generators?

Reply:
async done; the rest is too exotic for now!